cristin-organization/src/main/java/no/unit/nva/cristin/organization/common/client/v20230526/FetchCristinOrgClient20230526.java

```diff
@@ -0,0 +1,96 @@
+package no.unit.nva.cristin.organization.common.client.v20230526;
+
+import static java.util.Arrays.asList;
+import static no.unit.nva.HttpClientProvider.defaultHttpClient;
+import static no.unit.nva.cristin.model.Constants.CRISTIN_API_URL;
+import static no.unit.nva.cristin.model.Constants.CRISTIN_PER_PAGE_PARAM;
+import static no.unit.nva.cristin.model.Constants.NONE;
+import static no.unit.nva.cristin.model.Constants.PARENT_UNIT_ID;
+import static no.unit.nva.cristin.model.Constants.UNITS_PATH;
+import static no.unit.nva.cristin.model.JsonPropertyNames.DEPTH;
+import static no.unit.nva.cristin.model.JsonPropertyNames.IDENTIFIER;
+import static no.unit.nva.model.Organization.ORGANIZATION_CONTEXT;
+import static no.unit.nva.utils.UriUtils.createCristinQueryUri;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpResponse;
+import java.util.Map;
+import no.unit.nva.cristin.common.client.ApiClient;
+import no.unit.nva.cristin.common.client.FetchApiClient;
+import no.unit.nva.cristin.organization.dto.v20230526.UnitDto;
+import no.unit.nva.cristin.organization.dto.v20230526.mapper.OrganizationFromUnitMapper;
+import no.unit.nva.model.Organization;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+import nva.commons.apigateway.exceptions.BadGatewayException;
+import nva.commons.core.paths.UriWrapper;
+
+public class FetchCristinOrgClient20230526 extends ApiClient
+    implements FetchApiClient<Map<String, String>, Organization> {
+
+    public static final String ALL_RESULTS = "2000";
+
+    public FetchCristinOrgClient20230526() {
+        this(defaultHttpClient());
+    }
+
+    public FetchCristinOrgClient20230526(HttpClient client) {
+        super(client);
+    }
+
+    /**
+     * Fetch one organization matching given query criteria. By specifying query param depth one can choose if the
+     * response should include sub organization tree downwards or just have the upwards parent trail.
+     *
+     * @param params Map containing verified query parameters
+     */
+    @Override
+    public Organization executeFetch(Map<String, String> params) throws ApiGatewayException {
+        var identifier = params.get(IDENTIFIER);
+        var fetchUri = getCristinUri(identifier);
+        var response = fetchGetResult(fetchUri);
+        if (wantsDepth(params)) {
```

Comment:
hasDepth?

Reply:
hasDepth is not entirely correct. We dont know yet that the organization has depth at this stage. But if the client has not specified he does not want depth, we assume wantsDepth == true and then do an extra request for fetching requested organization's subunits. We dont get everything in one request because of how upstream is modelled